bridges/modules/messages/src/lib.rs

```diff
@@ -644,6 +617,60 @@ pub mod pallet {
 		}
 	}
 
+	#[cfg(any(feature = "try-runtime", test))]
+	impl<T: Config<I>, I: 'static> Pallet<T, I> {
+		/// Ensure the correctness of the state of this pallet.
+		pub fn do_try_state() -> Result<(), sp_runtime::TryRuntimeError> {
+			Self::do_try_state_for_outbound_lanes()
+		}
+
+		/// Ensure the correctness of the state of outbound lanes.
+		pub fn do_try_state_for_outbound_lanes() -> Result<(), sp_runtime::TryRuntimeError> {
+			use sp_runtime::traits::One;
+			use sp_std::vec::Vec;
+
+			// collect unpruned lanes
+			let mut unpruned_lanes = Vec::new();
+			for (lane_id, lane_data) in OutboundLanes::<T, I>::iter() {
+				let Some(expected_last_prunned_nonce) =
+					lane_data.oldest_unpruned_nonce.checked_sub(One::one())
+				else {
+					continue;
+				};
+
+				// collect message_nonces that were supposed to be pruned
+				let mut unpruned_message_nonces = Vec::new();
+				let mut nonce_to_check = expected_last_prunned_nonce;
+				loop {
+					if OutboundMessages::<T, I>::contains_key(MessageKey {
+						lane_id,
+						nonce: nonce_to_check,
+					}) {
+						unpruned_message_nonces.push(nonce_to_check);
+					}
+					if let Some(new_nonce_to_check) = nonce_to_check.checked_sub(One::one()) {
```

Comment:
Nit: I would use a match here instead of if-else. Seems a bit more readable.

Reply:
I little bit simplified that to simple `for`, please take a look: https://github.com/paritytech/polkadot-sdk/pull/4944/commits/fb8d0ba860649be82cfef18c29778ebb540b0be4
also probably we don't need to iterate all the time from actual nonce to the zero, again and again, it will be waste of time, also probably this `oldest_unpruned_nonce` is not needed